### ISSUE TITLE
Output same word

### ISSUE THREAD

**Issue by dmuestc** [likejazz/llama3.cuda#5]
Hi, I download code and compile it, but got unexpected results.
environments: 
  OS: centos 7.5
  GPU: V100, 32G
  gcc: gcc-8

make -j 20
./runcuda "I have a dream"

The model outputs: "I have a dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream dream"

It's some weird.

**Comment by likejazz** [comment]
Unfortunately I don't have V100 so I can't test it, could you please print the whole logit value for debugging?

**Comment by dmuestc** [comment]
> Reference in n

yes, I print the logits for the first five value, they are [0.019237, 0.026642, 0.019238, 0.019239, 0.019238], in line 821.
when generate  function goes on, the logits  keep all the same. It seems that the new token has not participate in generation process.
For each iteration, the largest prob value is always 0.176558, so the decoded token keeps "dream".
my cuda version is 12.2, gcc is 8.5.

**Comment by saurabh-singh-rajput** [comment]
Same issue on my 3070, any fixes?
